website/docs/docs/build/udfs.md

```diff
@@ -0,0 +1,208 @@
+    AS (
+    REGEXP_CONTAINS(a_string, r'^[0-9]+$')
+    );
+    ```
+    </TabItem>
+
+    <TabItem value="Redshift">
+    ```sql
+    CREATE OR REPLACE FUNCTION repeat_n(s VARCHAR, n INTEGER)
+    RETURNS VARCHAR
+    VOLATILE
+    AS $$ SELECT repeat(s, n); $$
+    LANGUAGE SQL;
+    ```
+    </TabItem>
+    </Tabs>
+
+3. Reference the UDF in a model using the `{{ function(...) }}` macro. For example:
+
+    <File name="models/my_model.sql">
+
+    ```sql
+    select
+
+    maybe_positive_int_column,
+        {{ function('is_positive_int') }}(maybe_positive_int_column)
+
+    from {{ ref('a_model_i_like') }}
+    ```
+    </File>
+
+4. Run `dbt compile`. In the following example, the `{{ function('is_positive_int') }}` is replaced by the UDF name `udf_db.udf_schema.is_positive_int`.
+
+    <File name="models/my_model.sql">
+
+    ```sql
+    select
+
+    maybe_positive_int_column,
+	udf_db.udf_schema.is_positive_int(maybe_positive_int_column) as is_positive
+
+    from analytics.<dbt_schema>.a_model_i_like
+    ```
+    </File>
+
+    In your DAG, there should be a dependency between `is_positive_int` → `my_model` and a UDF node is created from the SQL and YAML definitions.
+
+After defining a UDF, you can update the SQL file that contains its function body (`is_positive_int.sql` in this example). When you rebuild, your changes will be applied everywhere the UDF is referenced. 
+
+## Using UDFs in unit tests
+
+You can use [unit tests](/docs/build/unit-tests) to validate models that reference UDFs. Before running unit tests, make sure the function exists in your warehouse. To ensure that the function exists for a unit test, run:
+
+```bash
+dbt build --select "+my_model_to_test" --empty
+```
+
+Following the example in [Defining UDFs in dbt](#defining-udfs-in-dbt), here's an example of a unit test that validates a model that calls a UDF:
+
+<File name="tests/test_is_positive_int.yml">
+
+```yml
+unit_tests:
+  - name: test_is_positive_int 
+    description: "Check my is_positive_int logic captures edge cases"
+    model: my_model
+    given:
+      - input: ref('a_model_i_like')
+        rows:
+          - { maybe_positive_int_column: 10 }
+          - { maybe_positive_int_column: -4 }
+          - { maybe_positive_int_column: +8 }
+          - { maybe_positive_int_column: 1.0 }
+    expect:
+      rows:
+        - { maybe_positive_int_column: 10,  is_positive: true }
+        - { maybe_positive_int_column: -4,  is_positive: false }
+        - { maybe_positive_int_column: +8,  is_positive: true }
+        - { maybe_positive_int_column: 1.0, is_positive: true }
+```
+</File>
+
+## Listing and selecting UDFs
+
+To list UDFs in your project, run `dbt list`. 
+
+To select UDFs when running a project, use the following commands:
+
+- `dbt run --resource_type function` &mdash; Use this command to only run UDFs in your project.
+- `dbt run --select resource_type:function` &mdash; Use this command to reinitialize all UDFs in your project.
```

Comment:
@QMalcolm does both `--select resource_type:function` and `--resource_type function` work for `build`? i thought `--resource-type` was just for `list`?

https://docs.getdbt.com/reference/node-selection/methods#resource_type

```suggestion
- `dbt build --select "resource_type:function"` &mdash; Use this command to build all UDFs in your project.
```

Reply:
`--resource-type` _does_ work for build. Just tested it both in cloud and in core. Of note the documentation says `--resource_type` and it should be `--resource-type`